modules/backendworker/backendworker.go

```diff
@@ -0,0 +1,254 @@
+package backendworker
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/backoff"
+	"github.com/grafana/dskit/services"
+	"github.com/grafana/dskit/user"
+	backendscheduler_client "github.com/grafana/tempo/modules/backendscheduler/client"
+	"github.com/grafana/tempo/modules/overrides"
+	"github.com/grafana/tempo/modules/storage"
+	"github.com/grafana/tempo/pkg/model"
+	"github.com/grafana/tempo/pkg/tempopb"
+	tempo_util "github.com/grafana/tempo/pkg/util"
+	"github.com/grafana/tempo/pkg/util/log"
+	"github.com/grafana/tempo/tempodb/backend"
+)
+
+const (
+	reasonCompactorDiscardedSpans = "trace_too_large_to_compact"
+)
+
+type BackendWorker struct {
+	services.Service
+
+	cfg              Config
+	store            storage.Store
+	overrides        overrides.Interface
+	backendScheduler tempopb.BackendSchedulerClient
+
+	workerID string
+}
+
+// var tracer = otel.Tracer("modules/backendworker")
+
+func New(cfg Config, schedulerClientCfg backendscheduler_client.Config, store storage.Store, overrides overrides.Interface) (*BackendWorker, error) {
+	s := &BackendWorker{
+		cfg:       cfg,
+		store:     store,
+		overrides: overrides,
+	}
+
+	workerID, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+	s.workerID = workerID
+
+	schedulerClient, err := backendscheduler_client.New(cfg.BackendSchedulerAddr, schedulerClientCfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create backend scheduler client: %w", err)
+	}
+	s.backendScheduler = schedulerClient
+
+	s.Service = services.NewBasicService(s.starting, s.running, s.stopping)
+	return s, nil
```

Comment:
should return an error in the event of bad config. i.e. 0s in backoff setting or something

Reply:
Good call, I'll add some config validation.